packages/utilities/src/inferLanguages.ts

```diff
@@ -128,6 +130,17 @@ function getLanguage(languages: Map<string, Language>, languageName: string) {
     return language
 }
 
+function getPropertyName(key: string) {
```

Comment:
Getting better, although this heuristic probably fails quite easily, e.g. if there are less than 2 separators. Would it be an idea to write some unit tests for the `getPropertyName` function (which I'd name `deriveLikelyPropertyName` or something like that to indicate it's an educated guess)?